fe/fe-core/src/main/java/com/starrocks/sql/StatementPlanner.java

```diff
@@ -39,15 +39,29 @@
 import com.starrocks.sql.optimizer.transformer.RelationTransformer;
 import com.starrocks.sql.plan.ExecPlan;
 import com.starrocks.sql.plan.PlanFragmentBuilder;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 import java.util.List;
 import java.util.Map;
 import java.util.stream.Collectors;
 
 public class StatementPlanner {
+    private static final Logger LOG = LogManager.getLogger(StatementPlanner.class);
+
     public ExecPlan plan(StatementBase stmt, ConnectContext session) throws AnalysisException {
+        if (session.getSessionVariable().isEnableOptimizerTraceLog()) {
+            if (stmt instanceof QueryStatement) {
+                LOG.info("[TRACE QUERY {}] After Parse\n{}\n", session.getQueryId(), stmt.toString());
+            }
+        }
```

Comment:
I think you can define a method `getOptimizeTraceLevel()` in session, and return `INFO` if enable optimizer trace log else return `DEBUG`/`TRACE`, then you can code:
```
LOG.log(getOptimizeTraceLevel(), 'xxxxxxxxxxxxxxxx');
```

or another ways, define a OptimizerTracer class, to support log interface and replace your `OptimizerTraceInfo` class

Reply:
> I think you can define a method `getOptimizeTraceLevel()` in session, and return `INFO` if enable optimizer trace log else return `DEBUG`/`TRACE`, then you can code:
> 
> ```
> LOG.log(getOptimizeTraceLevel(), 'xxxxxxxxxxxxxxxx');
> ```
> 
> or another way, define a OptimizerTracer class, to support log interface and replace your `OptimizerTraceInfo` class I perfer way 2

Yes, unifying the output interface will be clearer and reduce a log of judgment logic, I will fix it

Reply:
> I think you can define a method `getOptimizeTraceLevel()` in session, and return `INFO` if enable optimizer trace log else return `DEBUG`/`TRACE`, then you can code:
> 
> ```
> LOG.log(getOptimizeTraceLevel(), 'xxxxxxxxxxxxxxxx');
> ```
> 
> or another way, define a OptimizerTracer class, to support log interface and replace your `OptimizerTraceInfo` class I perfer way 2

I think it is necessary to keep `OtpmizerTraceInfo` class to facilitate the collection of trace information, I will add a new class to deal with the output problem